pkg/ddevapp/config.go

```diff
@@ -722,6 +723,18 @@ func (app *DdevApp) FixObsolete() {
 		}
 	}
 
+	// Remove old ~/.ddev_mutagen_data_directory directory
+	legacyMutagenDataDir := filepath.Join(homedir.Get(), ".ddev_mutagen_data_directory")
+	if fileutil.IsDirectory(legacyMutagenDataDir) {
+		originalMutagenDataDir := os.Getenv("MUTAGEN_DATA_DIRECTORY")
+		_ = os.Setenv("MUTAGEN_DATA_DIRECTORY", legacyMutagenDataDir)
+		StopMutagenDaemon()
+		_ = os.Setenv("MUTAGEN_DATA_DIRECTORY", originalMutagenDataDir)
+		err := os.RemoveAll(legacyMutagenDataDir)
```

Comment:
This will fail certainly on Windows, maybe elsewhere if the daemon is still running. Have we killed off the daemon here? It's probably OK with the existing logic (tried and failed notification) but worth talking about.

Reply:
I thought the daemon would be stopped here because DDEV asks for poweroff during the upgrade.